client/state/posts/middleware.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { isNumber, toArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { TERMS_RECEIVE } from 'state/action-types';
+import { getEditedPost } from 'state/posts/selectors';
+import { editPost } from 'state/posts/actions';
+
+/**
+ * Dispatches editPost when a new term has been added
+ * that also has a postId on the action
+ *
+ * @param  {Function} dispatch  Dispatch method
+ * @param  {Object}   state     Global state tree
+ * @param  {Object}   action    Action object
+ */
+export function onTermsReceive( dispatch, state, action ) {
+	const { postId, taxonomy, terms, siteId } = action;
+	const post = getEditedPost( state, siteId, postId );
+	const newTerm = terms[ 0 ];
```

Comment:
I guess currently it's not possible to receive more than a single term to be associated with an edited post, but perhaps in future iterations we ought consider not explicitly pulling the first entry and rather accommodate an arbitrary number of terms received.